### ISSUE TITLE
Support for PUT/PATCH/DELETE verbs

### ISSUE THREAD

**Issue by qpfiffer** [warmcat/libwebsockets#228]
I have some patches in a fork that I'd love to see in mainline, but I don't want to just dump it in here with some discussion.

You can see the differences [here.](https://github.com/qpfiffer/libwebsockets/compare/warmcat:master...master) If theres any interest or feed back I'll make a pull request.

**Comment by JoakimSoderberg** [comment]
You don't want a discussion? Or did you mean: "I don't want to just dump it in here **without** some discussion **first**."

The point with pull requests is to show the changes made and to use a discussion thread. It makes it easier, since individual lines and such can be commented. And the pull request gets updated if you push new changes to your branch.

Also, it is important, since if you create a pull request all the tests will run on your changes automatically so it's easy to see if compilation breaks on some platform.

In short: Create a pull request :)

**Comment by qpfiffer** [comment]
Will do.

**Comment by JoakimSoderberg** [comment]
@qpfiffer Hey! Are you going to create that pull request? I didn't mean to scare you away :)

**Comment by JoakimSoderberg** [comment]
@qpfiffer Actually it was possible for me to do it for you in #230, didn't know that... Any further discussion can continue there
